autoPyTorch/datasets/base_dataset.py

```diff
@@ -82,6 +97,17 @@ def __init__(
         # or for augmentation
         self.transform = transforms
 
+    @staticmethod
+    def is_small_dataset(data_tensors: Iterable[BASE_DATASET_INPUT]) -> bool:
+        """
+        Determine wether the whole dataset can be preprocessed in memory.
+        :param data_tensors: An iterable holding tensors used to infer the dataset size
+        """
+        available_memory_bytes = psutil.virtual_memory().available
```

Comment:
Do you think we can take into account the fit dictionary memory usage? For instance, if in a cluster you have 100 Gb of memory, but you are only allowed few Gb, psutil will always return 100Gb, so any dataset will be small.

Reply:
Mhh true, I guess the API could pass that to the dataset and call the method again.

Another way might be to just `try - except` do something with the dataset and catch a memory error, maybe I'll add both